src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py

```diff
@@ -118,8 +123,12 @@ def _search_role_definitions(definitions_client, name, scope, custom_role_only=F
     return roles
 
 
-def create_role_assignment(cmd, role, assignee, resource_group_name=None, scope=None):
-    return _create_role_assignment(cmd.cli_ctx, role, assignee, resource_group_name, scope)
+def create_role_assignment(cmd, role, assignee=None, assignee_object_id=None, resource_group_name=None, scope=None):
+    if bool(assignee) == bool(assignee_object_id):
+        raise CLIError('usage error: --assignee STRING | --assignee-object-id GUID')
```

Comment:
Is there a reason we couldn't simply accept a string or GUID for assignee? This seems like an Xplat anti-pattern.

Reply:
The problem is GUID can be used as assignee names, so w/o graph query, we would not know. For people login with service principal, they need to have the option to bypass the graph query.
I have proposed to the fallback to object id as long as it is a guid, but it is possible we could assign by mistake when service principal 2's object id happens to be the service principal 1's name